test/Microsoft.OData.Cli.Tests/CodeGeneration/ODataCliCodeGenerationTests.cs

```diff
@@ -0,0 +1,475 @@
+﻿//-----------------------------------------------------------------------------------
+// <copyright file="ODataCliCodeGenerationTests.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------------------
+
+using System.CommandLine;
+using System.CommandLine.Parsing;
+
+namespace Microsoft.OData.Cli.Tests.CodeGeneration
+{
+    public class ODataCliCodeGenerationTests
+    {
+        private readonly GenerateCommand generateCommand;
+        private static readonly string metadataUri;
+        private static readonly string lowerCamelCaseMetadataUri;
+        private static readonly string outputDir;
+
+        static ODataCliCodeGenerationTests()
+        {
+            metadataUri = Path.Combine(Environment.CurrentDirectory, "CodeGeneration\\Artifacts\\SampleServiceV4.xml");
+            lowerCamelCaseMetadataUri = Path.Combine(Environment.CurrentDirectory, "CodeGeneration\\Artifacts\\SampleServiceV4LowerCamelCase.xml");
+            outputDir = Path.Combine(Path.GetTempPath(), "twuf0gqr.d25");
+        }
+
+        public ODataCliCodeGenerationTests()
+        {
+            this.generateCommand = new GenerateCommand();
+            if (Directory.Exists(outputDir))
+            {
+                // Empty directory
```

Comment:
It wasn't clear that this a verb and not an adjective, which was confusing for me at first.
```suggestion
                // Let's empty directory
```